backend/queue/queue.go

```diff
@@ -291,14 +353,14 @@ func (q *Queue) tryDelete(ctx context.Context, kv *mvccpb.KeyValue) (types.Queue
 	key := string(kv.Key)
 
 	// generate a new key name
-	uName, err := q.uniqueName()
+	seq, err := q.timeStamp()
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("error deleting queue item: %s", err)
 	}
-	uKey := path.Join(q.inFlight, uName)
+	uKey := path.Join(q.inFlightPrefix(), seq)
 
-	delCmp := clientv3.Compare(clientv3.ModRevision(key), "=", kv.ModRevision)
-	putCmp := clientv3.Compare(clientv3.ModRevision(uKey), "=", 0)
+	delCmp := clientv3.Compare(clientv3.Version(key), ">", 0)
+	putCmp := clientv3.Compare(clientv3.Version(uKey), "=", 0)
```

Comment:
What is the difference between `Version` and `ModRevision`? The [docs](https://godoc.org/github.com/coreos/etcd/clientv3#ModRevision) don't really help here.

Reply:
The distinction is a bit subtle and not well explained in the docs. There is an explainer issue here: https://github.com/coreos/etcd/issues/6518